Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1968,7 +1998,7 @@ namespace Babylon
 
         private:
             Napi::ObjectReference m_jsSession{};
-            std::array<Napi::ObjectReference, 2> m_jsRenderTargetTextures;
+            std::array<Napi::ObjectReference, 3> m_jsRenderTargetTextures;
```

Comment:
nit: If this no longer has a guaranteed two elements (which I guess probably hasn't been true for a while now), should it be a vector now instead of an array?

Reply:
Do iOS and Android use more than one render target texture? Right now, it looks like we pre-construct these textures when creating the NativeRenderTargetProvider. We may be creating two more textures than needed when executing the iOS and Android scenarios.

It seems a bit unclear whether GetRenderTargetForEye is called based on information provided in the NativeRenderTargetProvider callback. This makes me hesitant in looking into dynamically creating render targets textures as needed. It feels a bit beyond the scope of this review.

Reply:
this should now be addressed